src/main/java/com/snowflake/kafka/connector/Utils.java

```diff
@@ -631,4 +643,130 @@ public static void updateConfigErrorMessage(Config result, String key, String ms
+        avroConverterConfig.getSchemaRegistryUrls(),
+        avroConverterConfig.getMaxSchemasPerSubject(),
+        Collections.singletonList(new AvroSchemaProvider()),
+        srConfig,
+        avroConverterConfig.requestHeaders());
+  }
+
+  @VisibleForTesting
+  public static Map<String, String> getSchemaFromSchemaRegistryClient(
+      final String topicName, final SchemaRegistryClient schemaRegistry, final String type) {
+    String subjectName = topicName + "-" + type;
+    SchemaMetadata schemaMetadata;
+    try {
+      schemaMetadata = schemaRegistry.getLatestSchemaMetadata(subjectName);
+    } catch (Exception e) {
+      throw SnowflakeErrors.ERROR_0012.getException();
+    }
+    Map<String, String> schemaMap = new HashMap<>();
+    if (schemaMetadata != null) {
+      AvroSchema schema = new AvroSchema(schemaMetadata.getSchema());
+      for (Schema.Field field : schema.rawSchema().getFields()) {
+        Schema fieldSchema = field.schema();
+        if (!schemaMap.containsKey(field.name())) {
+          switch (fieldSchema.getType()) {
+            case BOOLEAN:
+              schemaMap.put(field.name(), "boolean");
+              break;
+            case BYTES:
+              schemaMap.put(field.name(), "binary");
+              break;
+            case DOUBLE:
+              schemaMap.put(field.name(), "double");
+              break;
+            case FLOAT:
+              schemaMap.put(field.name(), "float");
+              break;
+            case INT:
+              schemaMap.put(field.name(), "int");
+              break;
+            case LONG:
+              schemaMap.put(field.name(), "number");
+              break;
+            case STRING:
+              schemaMap.put(field.name(), "string");
+              break;
+            case ARRAY:
+              schemaMap.put(field.name(), "array");
+              break;
+            default:
+              schemaMap.put(field.name(), "variant");
+          }
+        }
+      }
+    }
+    return schemaMap;
+  }
+
+  /**
+   * From the connector config extract whether the avro value converter is used
+   *
+   * @param connectorConfig
+   * @return whether the avro value converter is used
+   */
+  public static boolean usesAvroValueConverter(final Map<String, String> connectorConfig) {
+    List<String> validAvroConverter = new ArrayList<>();
+    validAvroConverter.add("io.confluent.connect.avro.AvroConverter");
+    validAvroConverter.add("com.snowflake.kafka.connector.records.SnowflakeAvroConverter");
+    validAvroConverter.add("io.confluent.encryption.connect.SecuredAvroConverter");
+    // the last one in doubt
+    if (connectorConfig.containsKey(SnowflakeSinkConnectorConfig.VALUE_CONVERTER_CONFIG_FIELD)) {
+      String valueConverter =
+          connectorConfig.get(SnowflakeSinkConnectorConfig.VALUE_CONVERTER_CONFIG_FIELD);
+      return validAvroConverter.contains(valueConverter);
+    } else {
+      return false;
+    }
```

Comment:
Umm, I am not entirely convinced this is true. 
In case the value converter is not used, it is still possible they have avro converter since it can be defined at connector JVM level and not at specific config level. 

So returning false is kind of harmful here. 

In this case (i.e instead of returning false) we can check if we can fetch schema from schema registry. (Which is not possible until we get records)
Or fetch schema registry URL- again can be set at worker level. 

The documentation says it can be defined at worker level and can be overridden by connector. 
https://docs.confluent.io/home/connect/self-managed/userguide.html#configuring-key-and-value-converters

I dont see a way to fetch this properties at worker level. 

So we can either enforce this at connector level or we can do hacky way which is to wait till we see the record. 

My preference is we should ask customers to provide this at connector level too since they will anyways need to give us schema registry URL anyways. 
so on top of schematization, we will need value and key converters with url.

Reply:
If it returns false we will create a table with only the RECORD_METADATA column, and the schema evolution should handle the things since then, which is not bad, right?

I also prefer enforcing it at the connector level. May I add this to the documentation JIRA that the user need to set these fields on the connector config if they want to use this feature?

Reply:
> If it returns false we will create a table with only the RECORD_METADATA column, and the schema evolution should handle the things since then, which is not bad, right?
Sounds good to me! Please mention this in the else case. 

Yes please add this to documentation. Thank you!

Reply:
Not sure if I understand this thread, why can't we try to get it from the worker level if we don't see it in the configuration file?